pkg/engine/lifeycletest/pulumi_test.go

```diff
@@ -2192,3 +2192,167 @@ func TestComponentOutputs(t *testing.T) {
 	}
 	p.Run(t, nil)
 }
+
+// Test calling a method.
+func TestSingleComponentMethodDefaultProviderLifecycle(t *testing.T) {
+	var urn resource.URN
```

Comment:
This trips me up why this would be a shared variable, just abbreviating for convenience? 

My expectation would be that executing Call would identify the resource instance somehow by passing a unique identifier such as URN somewhere, instead we have:

```
	outs, _, _, err := monitor.Call("pkgA:m:typA/methodA", resource.PropertyMap{
			"name": resource.NewStringProperty("Alice"),
		}, "", "")
```

And the mock ProviderLoader relies on the "global" `var urn` to find the resource. That deviates from the usage in actual prod code in this manner, or am I misunderstanding something? Thanks.